### ISSUE TITLE
PromiseConstructor error message is not using the correct format for newer versions of ECMAScript

### ISSUE THREAD

**Issue by Vboivin** [microsoft/TypeScript#40929]
**TypeScript Version:**  Nightly

**Search Terms:**
- Promises
- Error Messages
- Compiler

**Code**

```ts
// es2018 feature on Promise Type (correct error behavior)
new Promise(() => {}).finally();

// es2020 feature on PromiseConstructor Type (wrong error behavior)
Promise.allSettled([]);

// esnext feature on PromiseConstructor Type (wrong error behavior)
Promise.any([]);
```

**Expected behavior:**
- `Property 'allSettled' does not exist on type 'PromiseConstructor'. Do you need to change your target library? Try changing the lib compiler option to 'es2020' or later.`
- `Property 'any' does not exist on type 'PromiseConstructor'. Do you need to change your target library? Try changing the lib compiler option to 'esnext' or later.`

**Actual behavior:**
- `Property 'allSettled' does not exist on type 'PromiseConstructor'`.
- `Property 'any' does not exist on type 'PromiseConstructor'`.

**Playground Link:** 
- https://www.typescriptlang.org/play?ts=4.1.0-dev.20201003#code/PTAEEsDsGMHsCd4FNoBdQCMkAsCGA3cWAV3gCgAFeWAW3AGckA6XAG1YGUlVVWkATABQBtALoBKANyVqdRi0gBPEROlkQoOIhToseQiXKQkAd1BVaDJIMHjQAXgB8oAN4BfcUwBmUNq2VSQA

**Related Issues:** 
- #40320

**How To Resolve**
- Replace key at https://github.com/microsoft/TypeScript/blob/master/src/compiler/utilities.ts#L559 by PromiseConstructor
- Replace key at https://github.com/microsoft/TypeScript/blob/master/src/compiler/utilities.ts#L599 by PromiseConstructor
- Replace key at https://github.com/microsoft/TypeScript/blob/master/src/compiler/utilities.ts#L606 by PromiseConstructor
- Change test at https://github.com/microsoft/TypeScript/blob/61aadc4ce2ed079d8ac4ec2bc7a6ab22f43ff6d3/tests/cases/compiler/doYouNeedToChangeYourTargetLibraryES2016Plus.ts#L38
- Change test at https://github.com/microsoft/TypeScript/blob/61aadc4ce2ed079d8ac4ec2bc7a6ab22f43ff6d3/tests/cases/compiler/doYouNeedToChangeYourTargetLibraryES2016Plus.ts#L44

**Comment by DanielRosenwasser** [comment]
Thanks for catching that 🙂
